### ISSUE TITLE
False positive on Naming/VariableNumber (conflicts with Lint/UnusedMethodArgument)

### ISSUE THREAD

**Issue by halostatue** [rubocop-hq/rubocop#6709]
`Naming/VariableNumber` cop indicates that `normalcase` should be used for numbered variables. The variables here exist explicitly to satisfy an API contract caused by inheritance, and I neither care nor need the variables (I am calling `super nil`). I could use letters to avoid this cop, but this is a special case of two cops not getting along (e.g., `Lint/UnusedMethodArgument`).

```
class Foo < ::Logger
  def initialize(_1, _2 = nil, _3 = nil)
    super nil
    @logdev = WarnLogDevice.new
    @formatter = ->(_s, _d, _p, m) { m }
  end
end
```

--------

## Expected behavior

No error is reported.

## Actual behavior

Received the report `Naming/VariableNumber: Use normalcase for variable numbers.`

## Steps to reproduce the problem

Run RuboCop over the code above.

## RuboCop version

0.63.1 (using Parser 2.5.1.0, running on ruby 2.6.0 x86_64-darwin18)

**Comment by Drenmi** [comment]
Thanks for your report. 🙇 

There are three styles commonly seen for unused parameters.

**1. Naming the parameters meaningfully, but prepend underscore:**

```ruby
def initialize(_foo, _bar = nil, _baz = nil)
```

This preserves the entire method signature.

**2. Using just underscores:**

```ruby
def initialize(_, _ = nil, _ = nil)
```

This preserves just the arity of the method signature.

**3. Using just a splat:**

```ruby
def initialize(*)
```

This obscures the entire method signature.

---

It doesn't seem that numbering the unused variables offers any value. I'm hesitant to increase the complexity of the variable numbering cop to account for this edge case.

**Comment by halostatue** [comment]
Regardless, the error reported is incorrect and confusing. I ultimately don’t care what is done with this as I turn off most cops, but the report doesn’t actually suggest a useful fix in this case.
